tests: uart: esp32: Test configuration

raffarost opened the pull request:
Add file configs for ESP32 devices for CI testing.
Get baudrate from device tree for `uart_basic_api` and `uart_elementary` tests.
Fix esp32 uart driver function `uart_get_config` to return standard baudrate value.

Commit:
tests: uart_basic_api: uart_elementary: Get baudrate from device tree

Add option (BAUDRATE_FROM_DTS) to get baudrate value from
device tree, for devices which differ from standard (115200).

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: serial: esp32: Standard baudrate on config_get

Return standard baudrate on a uart_config_get call instead of
rough calculated numbers.

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
tests: uart_elementary: esp32: Add test configs

Add test configuration for ESP32 devices.

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

sylvioalves commented on tests/drivers/uart/uart_elementary/socs/esp32s3_procpu.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_DUAL_UART_TEST=y
```
I see that `ESP32-S2` and all `riscv` SoCs don´t have `CONFIG_DUAL_UART_TEST=y` set. Is that correct?

dcpleung commented on tests/drivers/uart/uart_basic_api/src/test_uart_config.c:
```diff
@@ -25,13 +25,13 @@
+	.parity = UART_CFG_PARITY_NONE,
+	.stop_bits = UART_CFG_STOP_BITS_1,
+	.data_bits = UART_CFG_DATA_BITS_8,
+	.flow_ctrl = UART_CFG_FLOW_CTRL_NONE};
```
Please keep the closing bracket on another line.

dcpleung commented on tests/drivers/uart/uart_basic_api/src/test_uart_config_wide.c:
```diff
@@ -25,13 +25,13 @@
+	.parity = UART_CFG_PARITY_NONE,
+	.stop_bits = UART_CFG_STOP_BITS_1,
+	.data_bits = UART_CFG_DATA_BITS_9,
+	.flow_ctrl = UART_CFG_FLOW_CTRL_NONE};
```
Ditto for closing bracket.

dcpleung reviewed (dismissed):
Since you are amending the changes, please leave the whitespace changes out of functional changes. This will make future reviewing (and git log/show) of code easier to understand.

raffarost commented:
hi @dcpleung, in the last commits I reverted the non-functional changes for the `elementary` test and added a Kconfig to allow configuring a value for the mismatch case. It could be fixed to 9600 as it was before, but since we are getting baudrate from the device tree, it made sense to let it be configurable as well.

wmrsouza commented on tests/drivers/uart/uart_elementary/Kconfig:
```diff
@@ -8,4 +8,11 @@ config DUAL_UART_TEST
+	default 9600
+	help
+	  UART baudrate for DUT_AUX that is intentionally different from DUT.
+	  Used when SETUP_MISMATCH_TEST is enabled.
```
Do you mind making UART_BAUDRATE_MISMATCH explicitly dependent on SETUP_MISMATCH_TEST?

raffarost replied:
good idea

raffarost commented:
@dcpleung @nashif PTAL

sylvioalves commented:
@dcpleung @kartben ping.

kartben commented:
Co-assigning to @sylvioalves as platform owner and due to the nature of the changes. 
@dcpleung is on leave but I'm confident his previous NACK has been addressed.
